src/serac/physics/state/finite_element_vector.cpp

```diff
@@ -150,4 +150,19 @@ double min(const FiniteElementVector& fe_vector)
   return global_min;
 }
 
+double innerProduct(const FiniteElementVector& v1, const FiniteElementVector& v2)
+{
+  SLIC_ERROR_IF(
+      v1.Size() != v2.Size(),
+      axom::fmt::format("Finite element vector of size '{}' can not inner product with another vector of size '{}'",
+                        v1.Size(), v2.Size()));
+  SLIC_ERROR_IF(v1.comm() != v2.comm(),
```

Comment:
To be really pedantic, you could also check whether the two vectors are written against the same `mfem::(Par)FinieteElementSpace` - you could imagine scenarios where two different spaces on the same communicator coincidentally have the same sizes, but you're doing a non-nonsensical inner product.